src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,243 @@
+package edu.berkeley.ce.rockslicing
+
+import scala.annotation.tailrec
+
+/**
+  * Selects joints from the input joint set that will be used to seed the RDD in order to maintain
+  * approximately balanced loads among processors.
+  */
+object SeedJointSelector {
+
+  /**
+    * Searches though sequence of joint sets to find best seed joints
+    *
+    * @param jointSets Seq of Seq's of Joints representing input joint sets
+    * @param inputVolume Block representing input rock volume
+    * @param numProcessors Number of processors to be used in the analysis
+    * @return Returns a Seq containing the seed joints that best divide the input volume. If unable to find
+    *         satisfactory seed joints, returns None.
+    */
+  @tailrec
+  def searchJointSets(jointSets: Seq[Seq[Joint]], inputVolume: Block,
+                      numProcessors: Int): Option[Seq[Joint]] = {
+    if (jointSets.isEmpty) {
+      println("ERROR: Unable to find seed joints")
+      None
+    } else if (jointSets.head(0).shape.isEmpty) {
+      // Check that joint set is persistent
+      val seedJointOption = findSeedJoints(jointSets.head, inputVolume, numProcessors, inputVolume.volume)
+
+      if (seedJointOption.isEmpty) {
+          searchJointSets(jointSets.tail, inputVolume, numProcessors)
+      } else {
+        seedJointOption
```

Comment:
Here `seedJointOption` won't be `None` necessarily. This is the case when it's `Some` and we can stop the recursion and return. When it is `None`, we move on to the next joint set to check if that one yields satisfactory seed joints. 
I've modified the code to throw an exception instead of returning `None`. I think your suggestion to do this makes the code much cleaner and clearer.